Digital_Signature/port-based/digital-signature-port-access.go

```diff
@@ -0,0 +1,1005 @@
+	DIGITAL_SIGNATURE_UNABLE_TO_VERIFY_SIGNATURE_INTERNAL_ERROR = "Was unable to verify the signature created.  Please contact LEAF TEAM."
+	DIGITAL_SIGNATURE_UNABLE_TO_VERIFY_SIGNATURE_PIV            = "Unable to verify signature with PIV Card.  Please ensure correct one is inserted."
+	DIGITAL_SIGNATURE_UNABLE_TO_VERIFY_SIGNATURE_CERT           = "Unable to verify signature with supplied public cert."
+	DIGITAL_SIGNATURE_CREATED                                   = "Successfully created a digital signature for document."
+	INPUT_SIGNATURE_CORRUPT                                     = "The string entered for the digital signature is incorrect or corrupt."
+	INPUT_CERTIFICATE_CORRUPT                                   = "The string entered for the public certificate is incorrect or corrupt."
+	VERIFY_SIGNATURE_ERROR_GENERIC                              = "Error occured with verification proccess.  Ensure running on Windows."
+	VERIFY_SIGNATURE_FALSE                                      = "The signature was not able to verified for the document and public certificate provied."
+	SIGNATURE_VERIFIED_WITH_PUBLIC_CERT                         = "The digital signature has been verified for the document entered with the public certificate."
+	SIGNATURE_VERIFIED_WITH_PIV_CARD                            = "The digital signature has been verified for the document entered with the PIV card."
+	DATA_TO_BE_SIGNED_TOO_SMALL                                 = "The data to be signed was empty."
+	DATA_TO_BE_SIGNED_TOO_BIG                                   = "The data being requested to sign is over 1 Mb."
+	SIGNATURE_FIELD_EMPTY                                       = "The signature field is empty."
+	CARDCERT_FIELD_EMPTY                                        = "The certificate field is empty."
+)
+
+// Windows processes needed for operation of encryption and decryption
+var (
+	ncrypt                                = syscall.NewLazyDLL("ncrypt.dll")
+	procNCryptFreeObject                  = ncrypt.NewProc("NCryptFreeObject")
+	crypt32                               = syscall.NewLazyDLL("crypt32.dll")
+	procCertOpenStore                     = crypt32.NewProc("CertOpenStore")
+	procCertFindCertificateInStore        = crypt32.NewProc("CertFindCertificateInStore")
+	procCertFreeCertificateContext        = crypt32.NewProc("CertFreeCertificateContext")
+	procCryptImportPublicKeyInfo          = crypt32.NewProc("CryptImportPublicKeyInfo")
+	modAdvapi32                           = windows.NewLazySystemDLL("advapi32.dll")
+	procCryptCreateHash                   = modAdvapi32.NewProc("CryptCreateHash")
+	procCryptHashData                     = modAdvapi32.NewProc("CryptHashData")
+	procCryptDestroyHash                  = modAdvapi32.NewProc("CryptDestroyHash")
+	procCryptGetHashParam                 = modAdvapi32.NewProc("CryptGetHashParam")
+	procCryptSignHashW                    = modAdvapi32.NewProc("CryptSignHashW")

```

Comment:
`CryptSignHashW` appears to be a deprecated API, and a non-deprecated API should be used instead:

![image](https://github.com/user-attachments/assets/ea5f1023-5635-440d-ba91-c550be0f8dc7)
https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-cryptsignhashw

Reply:
Yes.  I'm aware of this.  But I was running into issues with Windows 11 using the NG API's.  It was failing even on basic "Are you there?" calls.  This version is to get this service provided and then I was going to dig deeper into why Windows 11 was failing when accessing those api's.  At first I thought it might be because I was using Windows 11 Home, but that doesn't appear to be the case.